Mathlib/Analysis/InnerProductSpace/Defs.lean

```diff
@@ -470,16 +481,72 @@ def toNormedAddCommGroup : NormedAddCommGroup F :=
       eq_zero_of_map_eq_zero' := fun _ hx =>
         normSq_eq_zero.1 <| (sqrt_eq_zero inner_self_nonneg).1 hx }
 
+section
+
 attribute [local instance] toNormedAddCommGroup
 
-/-- Normed space structure constructed from an `InnerProductSpace.Core` structure -/
-def toNormedSpace : NormedSpace 𝕜 F where
-  norm_smul_le r x := by
-    rw [norm_eq_sqrt_re_inner, inner_smul_left, inner_smul_right, ← mul_assoc]
-    rw [RCLike.conj_mul, ← ofReal_pow, re_ofReal_mul, sqrt_mul, ← ofReal_normSq_eq_inner_self,
-      ofReal_re]
-    · simp [sqrt_normSq_eq_norm, RCLike.sqrt_normSq_eq_norm]
-    · positivity
+/-- Normed space core structure constructed from an `InnerProductSpace.Core` structure -/
+lemma toNormedSpaceCore : NormedSpace.Core 𝕜 F where
+  norm_nonneg x := norm_nonneg x
+  norm_eq_zero_iff x := norm_eq_zero
+  norm_smul c x := by
+    letI : NormedSpace 𝕜 F := toNormedSpace
+    exact _root_.norm_smul c x
+  norm_triangle x y := norm_add_le x y
+
+end
+
+/-- In a topological vector space, if the unit ball of a continuous scalar product is von Neumann
+bounded, then the scalar product defines the same topology as the original one. -/
+lemma topology_eq
+    [tF : TopologicalSpace F] [IsTopologicalAddGroup F] [ContinuousConstSMul 𝕜 F]
+    (h : ContinuousAt (fun (v : F) ↦ cd.inner v v) 0)
+    (h' : IsVonNBounded 𝕜 {v : F | re (cd.inner v v) < 1}) :
+    tF = cd.toNormedAddCommGroup.toMetricSpace.toUniformSpace.toTopologicalSpace := by
```

Comment:
```suggestion
/-- In a topological vector space, if the unit ball of a continuous inner product is von Neumann
bounded, then the inner product defines the same topology as the original one. -/
lemma topology_eq
    [tF : TopologicalSpace F] [IsTopologicalAddGroup F] [ContinuousConstSMul 𝕜 F]
    (h : ContinuousAt (fun (v : F) ↦ cd.inner v v) 0)
    (h' : IsVonNBounded 𝕜 {v : F | re (cd.inner v v) < 1}) :
    tF = cd.toNormedAddCommGroup.toMetricSpace.toUniformSpace.toTopologicalSpace := by
```
In English I think it's a bit clearer like this.

Reply:
Thanks!